Update serverless docs with newer architecture (#8680)
kyma-project/kyma: docs/serverless/06-01-function-cr.md

| **status.conditions.status**             | Not applicable | Describes the status of processing the Function CR by the Function Controller. It can be `True` for success, `False` for failure, or `Unknown` if the CR processing is still in progress. If the status of all conditions is `True`, the overall status of the Function CR is ready.                                                                                                                                                                                                                                              |
| **status.conditions.type**               | Not applicable | Describes a substage of the Function CR processing. There are three condition types that a Function has to meet to be ready: `ConfigurationReady`, `BuildReady`, and `Running`. When displaying the Function status in the terminal, these types are shown under `CONFIGURED`, `BUILT`, and `RUNNING` columns respectively. All condition types can change asynchronously depending on the type of Function modification, but all three need to be in the `True` status for the Function to be considered successfully processed. |

> 

Reviewer: This part is missing:

```
### Status reasons

Processing of a Function CR can succeed, continue, or fail for one of these reasons:
```

| `JobFailed`                      | `BuildReady`         | The image with the Function's configuration could not be created due to an error.                                         |
| `JobCreated`                     | `BuildReady`         | The Kubernetes Job resource that builds the Function image was created.                                                   |
| `JobRunning`                     | `BuildReady`         | The Job is in progress.                                                                                                   |
| `JobsDeleted`                    | `BuildReady`         | Previous Jobs responsible for building Function images were deleted.                                                      |
| `JobFinished`                    | `BuildReady`         | The Job was finished and the Function's image was uploaded to the Docker Registry.                                        |
| `DeploymentCreated`              | `Running`            | A new Deployment referencing the Function's image was created.                                                            |

> | `DeploymentUpdated`              | `Running`            | The existing Deployment was updated after such changes as the Function's image, scaling parameters, variables, or labels. |

Reviewer: ```suggestion
| `DeploymentUpdated`              | `Running`            | The existing Deployment was updated after changing the Function's image, scaling parameters, variables, or labels. |
```

Reviewer: (I know it was already there, but looking at it now, I would put it this way ;) )

| `DeploymentFailed`               | `Running`            | The Function's Pod crashed or could not start due to an error.                                                            |

> | `DeploymentWaiting`              | `Running`            | The Function was deployed and is waiting for Deployment's readiness.                                                      |

Reviewer: ```suggestion
| `DeploymentWaiting`              | `Running`            | The Function was deployed and is waiting for the Deployment to be ready.                                                      |
```

> | `DeploymentReady`                | `Running`            | The Function was deployed and ready to serve.                                                                             |

Reviewer: ```suggestion
| `DeploymentReady`                | `Running`            | The Function was deployed and is ready.                                                                             |
```

Reviewer: "ready to serve" - you mean like pasta? ;) If you want to use "serve," you would rather need to explain what it serves like "deployment is ready to serve the traffic". Still, I wouldn't use it, unless you insist ;)

| `ServiceCreated`                 | `Running`            | A new Service referencing the Function's Deployment was created.                                                          |

> | `ServiceUpdated`                 | `Running`            | The existing Service was updated after needed changes.                                                                    |

Reviewer: ```suggestion
| `ServiceUpdated`                 | `Running`            | The existing Service was updated after applying required changes.                                                                    |
```

Reviewer: (definitely more common)

> | `ServiceError`                   | `Running`            | There are some problems with Kubernetes Service.                                                                         |

Reviewer: Yeah, that one is quite vague. So there is no particular reason but a given (human-readable) error pops up? Maybe sth like this:

```
The Kubernetes Service could not be created or updated due to an error.                                                          
```

Author: Same answer as with HPA, I'll add some example as to what can go wrong with them

> | `HorizontalPodAutoscalerCreated` | `Running`            | A new HorizontalPodScaler referencing the Function's deployment was created.                                              |

Reviewer: ```suggestion
| `HorizontalPodAutoscalerCreated` | `Running`            | A new HorizontalPodScaler referencing the Function's Deployment was created.                                              |
```

> | `HorizontalPodAutoscalerUpdated` | `Running`            | The existing HorizontalPodScaler was updated after needed changes.                                                        |

Reviewer: ```suggestion
| `HorizontalPodAutoscalerUpdated` | `Running`            | The existing HorizontalPodScaler was updated after applying required changes.                                                        |
```

> | `HorizontalPodAutoscalerError`   | `Running`            | There are some problems with HorizontalPodAutoscaler.                                                                     |

Reviewer: ```
The HorizontalPodAutoscaler could not be created or updated due to an error.                                                          
```

Author: Those problems are more connected with some incorrect config, not with creation

## Related resources and components

The Function custom resource relies on these Kubernetes resources:

| Resource                                                                                              | Description                                                                   |
| ----------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------- |

> | [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/)                             | Stores the source code and dependencies of the Function.                      |

Reviewer: ```suggestion
| [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/)                             | Stores the Function's source code and dependencies.                      |
```

> | [Job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/)              | Builds an image with the Function code in a runtime.                          |

Reviewer: ```suggestion
| [Job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/)              | Builds an image with the Function's code in a runtime.                          |
```

> | [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)                   | Serves the Function image as microservice.                                    |

Reviewer: ```suggestion
| [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)                   | Serves the Function's image as a microservice.                                    |
```

> | [Service](https://kubernetes.io/docs/concepts/services-networking/service/)                           | Exposes Function's Deployment as a network service inside Kubernetes cluster. |

Reviewer: ```suggestion
| [Service](https://kubernetes.io/docs/concepts/services-networking/service/)                           | Exposes the Function's Deployment as a network service inside the Kubernetes cluster. |
```

> | [HorizontalPodAutoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) | Automatically scales the number of Function's pods.                           |

Reviewer: ```suggestion
| [HorizontalPodAutoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) | Automatically scales the number of Function's Pods.                           |
```

| `DeploymentUpdated`              | `Running`            | The existing Deployment was updated after changing the Function's image, scaling parameters, variables, or labels. |
| `DeploymentFailed`               | `Running`            | The Function's Pod crashed or could not start due to an error.                                                     |
| `DeploymentWaiting`              | `Running`            | The Function was deployed and is waiting for the Deployment to be ready.                                           |
| `DeploymentReady`                | `Running`            | The Function was deployed and is ready.                                                                            |
| `ServiceCreated`                 | `Running`            | A new Service referencing the Function's Deployment was created.                                                   |
| `ServiceUpdated`                 | `Running`            | The existing Service was updated after applying required changes.                                                  |

> | `ServiceError`                   | `Running`            | There are some problems with Kubernetes Service.                                                                   |

Reviewer: ```suggestion
| `ServiceError`                   | `Running`            | There are issues with the Kubernetes Service that relate to configuration.                                                                   |
```

| `HorizontalPodAutoscalerCreated` | `Running`            | A new HorizontalPodScaler referencing the Function's Deployment was created.                                       |
| `HorizontalPodAutoscalerUpdated` | `Running`            | The existing HorizontalPodScaler was updated after applying required changes.                                      |

> | `HorizontalPodAutoscalerError`   | `Running`            | There are some problems with HorizontalPodAutoscaler.                                                              |

Reviewer: ```suggestion
| `HorizontalPodAutoscalerError`   | `Running`            | There are issues with the HorizontalPodAutoscaler that relate to configuration.                                                           |
```
